src/utils/child_process.ts

```diff
@@ -22,42 +22,58 @@ export interface ExecChildProcessResult<TOut = string> {
     readonly stderr: TOut;
 }
 
-export async function execChildProcess(command: string): Promise<ExecChildProcessResult> {
-    const r: Promise<ExecChildProcessResult> = new Promise((resolve, reject) => {
-        child_process.exec(command, {
+export async function execFile(command: string, args: string[], options: child_process.ExecFileOptions): Promise<ExecChildProcessResult> {
+    return new Promise<ExecChildProcessResult>((resolve, reject) => {
+        child_process.execFile(command, args, {
+            ...options,
             encoding: 'utf8',
         }, (error, stdout, stderr) => {
             if (!!error) {
                 reject(error);
                 return;
             }
-
             resolve({
                 stdout,
                 stderr,
             });
         });
     });
-    return r;
 }
 
+export async function run_rustup(rustup: string, args: string[], options: child_process.ExecFileOptions, useWSL?: boolean): Promise<ExecChildProcessResult> {
+    let command: string = rustup;
 
-export async function execFile(command: string, args: string[], options: child_process.ExecFileOptions): Promise<ExecChildProcessResult> {
-    return new Promise<ExecChildProcessResult>((resolve, reject) => {
-        child_process.execFile(command, args, {
-            ...options,
-            encoding: 'utf8',
-        }, (error, stdout, stderr) => {
-            if (!!error) {
-                reject(error);
-                return;
-            }
+    if (useWSL == true) {
+        ({ command: command, args: args } = modifyParametersForWSL(command, args));
+    }
 
-            resolve({
-                stdout,
-                stderr,
-            });
-        });
+    return execFile(command, args, options);
+}
 
-    });
+export function run_rustup_sync(rustup: string, args: string[], options: child_process.ExecFileOptions, useWSL?: boolean): Buffer {
```

Comment:
Could we please use camelCase and move running rustup to rustup.ts file?

Reply:
Changed snake_case to camelCase. Also moved code related to running a `rustup` to rustup.ts